src/axom/quest/examples/scattered_interpolation.cpp

```diff
@@ -0,0 +1,788 @@
+{
+  // Initialize the SLIC logger
+  axom::slic::SimpleLogger logger(axom::slic::message::Info);
+  // axom::slic::enableAbortOnWarning();
+
+  // Initialize default parameters and update with command line arguments:
+  Input params;
+  axom::CLI::App app {"Scattered interpolation over 2D or 3D point collection"};
+
+  try
+  {
+    params.parse(argc, argv, app);
+  }
+  catch(const axom::CLI::ParseError& e)
+  {
+    return app.exit(e);
+  }
+
+  /// Set up the input mesh
+
+  sidre::DataStore input_ds;
+
+  const std::string input_coords_name = "input_coords";
+  const std::string input_mesh_name = "input_mesh";
+  auto* input_pt_mesh_group =
+    input_ds.getRoot()->createGroup("input_point_mesh");
+  internal::blueprint::PointMesh inputMesh(input_pt_mesh_group,
+                                           input_coords_name,
+                                           input_mesh_name);
+
+  // Initialize the mesh points
+  switch(params.dimension)
+  {
+  case 2:
+    initializeInputMesh<2>(params, inputMesh);
+    break;
+  case 3:
+    initializeInputMesh<3>(params, inputMesh);
+    break;
+  }
+
+  // Write input mesh to file
+  {
+    std::string file = params.outputFile + "_mesh_after_generate";
+    SLIC_INFO(
+      axom::fmt::format("After generate points -- writing mesh file: '{}'", file));
+    inputMesh.saveMesh(file, params.outputProtocol);
+  }
+
+  /// Set up the query mesh
+
+  sidre::DataStore query_ds;
+  const std::string query_coords_name = "query_coords";
+  const std::string query_mesh_name = "query_mesh";
+  auto* query_pt_mesh_group =
+    query_ds.getRoot()->createGroup("query_point_mesh");
+  internal::blueprint::PointMesh queryMesh(query_pt_mesh_group,
+                                           query_coords_name,
+                                           query_mesh_name);
+
+  // Initialize the mesh points
+  switch(params.dimension)
+  {
+  case 2:
+    initializeQueryMesh<2>(params, queryMesh, inputMesh.getFieldNames());
+    break;
+  case 3:
+    initializeQueryMesh<3>(params, queryMesh, inputMesh.getFieldNames());
+    break;
+  }
+
+  /// Run the query
+  std::unique_ptr<quest::ScatteredInterpolation<2>> scattered_2d;
+  std::unique_ptr<quest::ScatteredInterpolation<3>> scattered_3d;
+
+  // Convert blueprint mesh from Sidre to Conduit
+  conduit::Node bp_input;
+  inputMesh.rootGroup()->createNativeLayout(bp_input);
+
+  conduit::Node bp_query;
+  queryMesh.rootGroup()->createNativeLayout(bp_query);
+
+  // Initialize the mesh points
+  switch(params.dimension)
+  {
+  case 2:
+    scattered_2d = std::unique_ptr<quest::ScatteredInterpolation<2>>(
+      new quest::ScatteredInterpolation<2>);
+    scattered_2d->buildTriangulation(bp_input, inputMesh.coordsName());
+    scattered_2d->locatePoints(bp_query, query_coords_name);
+
+    for(const auto& fld : inputMesh.getFieldNames())
+    {
+      scattered_2d->interpolateField(bp_query, query_coords_name, bp_input, fld, fld);
+    }
```

Comment:
Note: The bulk of this example relates to generating and loading mesh blueprint files.
The essential user-facing API is here:
* `buildTriangulation( input_mesh)  // as conduit node in blueprint format`
* `locatePoints( query_mesh) // as conduit node in blueprint format`
* // for each field
* `interpolateField( query_mesh, input_mesh, names of fields in query and input mesh)
  (the results are stored in the query mesh)

Based on a user-request, there will also be another function that, given a point in space, returns the vertex indices of the cell containing that point, and the barycentric coordinates of the point in its containing cell.